Reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in WebCore/html/track

cdumez opened the pull request:
#### a2319135abbdfa90bccd13139a4e45c68738d527
<pre>
Reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in WebCore/html/track
<a href="https://bugs.webkit.org/show_bug.cgi?id=284207">https://bugs.webkit.org/show_bug.cgi?id=284207</a>

Reviewed by Darin Adler.

* Source/WebCore/html/track/TextTrackCueList.cpp:
(WebCore::TextTrackCueList::activeCues):
(WebCore::TextTrackCueList::add):
(WebCore::TextTrackCueList::remove):
(WebCore::TextTrackCueList::updateCueIndex):
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCue::settingName):
* Source/WebCore/html/track/VTTRegion.cpp:
(WebCore::VTTRegion::scanSettingName):
* Source/WebCore/html/track/VTTScanner.h:
(WebCore::VTTScanner::scan): Deleted.
* Source/WebCore/html/track/WebVTTParser.cpp:
(WebCore::WebVTTParser::collectTimingsAndSettings):

Canonical link: <a href="https://commits.webkit.org/287510@main">https://commits.webkit.org/287510@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/34233ebdc95cc06613c8679bc1ca552df0e49b58

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/79947 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58944 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/33344 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/84461 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30921 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/82055 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/68008 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/7240 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/62482 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/20310 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/83014 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/52542 "Found 1 new test failure: media/audioSession/getUserMedia.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/72808 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42792 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/49884 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/26958 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/29383 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/71004 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/27439 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/85893 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/7163 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/5030 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/70753 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/7340 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/68648 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69998 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13994 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/12930 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12365 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/7127 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/12661 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/6974 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/10486 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8778 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/html/track/TextTrackCueList.cpp:
```diff
@@ -129,26 +129,26 @@ void TextTrackCueList::clear()
+    auto cueIndex = this->cueIndex(cue);
+    auto untilCuePosition = vectorSpan.first(cueIndex);
+    auto cuePosition = vectorSpan.subspan(cueIndex).begin();
+    auto afterCuePosition = vectorSpan.subspan(cueIndex + 1);
```
Now that these are spans I find the word `position` in these variable names actively confusing. Like how can a position be sorted? And nothing in `cuePosition` makes it clear that it's the cue and everything after it. Could clarify with 3 better names. Maybe `valuesBeforeCue`, `cueAndValuesAfterCue`, and `valuesAfterCue`, but I hope we can come up with shorter better names.

But I'm not sure, maybe we can just leave these as is.

cdumez replied:
Yes, I agree Position is confusing. I have renamed the variables.

cdumez committed:
Reduce use of WTF_ALLOW_UNSAFE_BUFFER_USAGE in WebCore/html/track
https://bugs.webkit.org/show_bug.cgi?id=284207

Reviewed by Darin Adler.

* Source/WebCore/html/track/TextTrackCueList.cpp:
(WebCore::TextTrackCueList::activeCues):
(WebCore::TextTrackCueList::add):
(WebCore::TextTrackCueList::remove):
(WebCore::TextTrackCueList::updateCueIndex):
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCue::settingName):
* Source/WebCore/html/track/VTTRegion.cpp:
(WebCore::VTTRegion::scanSettingName):
* Source/WebCore/html/track/VTTScanner.h:
(WebCore::VTTScanner::scan): Deleted.
* Source/WebCore/html/track/WebVTTParser.cpp:
(WebCore::WebVTTParser::collectTimingsAndSettings):

Canonical link: https://commits.webkit.org/287510@main
